controllers/authentication.controllers.ts

```diff
@@ -0,0 +1,262 @@
+import { Request, Response } from "express";
+import {
+  generateToken,
+  verifyToken
+} from "../shared/jwt-token.helpers";
+
+import Farm from "../models/Farm";
+import { IUser } from "../interfaces/user.interface";
+import User from "../models/User";
+import bcrypt from "bcrypt";
+import userMessage from "../messages/user.messages";
+
+// signUp a user
+async function signUp (req: Request, res: Response) {
+  try {
+    const userCheck = await User.findOne({ email: req.body?.email });
+
+    // if user email is already present
+    if (userCheck?._id) {
+      res.json({
+        message: userMessage.USER_CREATE_ACCOUNT_EXISTS,
+        data: null,
+        error: true
+      });
+
+      return;
+    }
+
+    const farm = await Farm.findById(req.body?.farmId).exec();
+    
+    // if invalid farmId was passed
+    if (!(farm?._id)) {
```

Comment:
if there is no farmId and there is a farmName let's create the farm.

Reply:
We aren't expecting a farm name while creation of a user... Should I pass a farm name

Reply:
We aren't passing farm name while creating a user... Should I add that field...

Reply:
Yes, pass the name, so we can create everything we need in this one endpoint. this endpoint is for our use only not the frontend, it should be as easy as possible for us to create the users.